tests/e2e_test.go

```diff
@@ -80,451 +29,18 @@ func TestE2E(t *testing.T) {
-	chainBNodeURIs = append(chainBNodeURIs, subnetBDetails.ValidatorURIs...)
-
-	log.Info("Created URIs for both subnets", "ChainAURIs", chainANodeURIs, "ChainBURIs", chainBNodeURIs, "blockchainIDA", blockchainIDA, "blockchainIDB", blockchainIDB)
-	log.Info("subnet ID", "subnetA", subnetADetails.SubnetID, "subnetB", subnetBDetails.SubnetID)
-
-	chainAWSURI := httpToWebsocketURI(chainANodeURIs[0], blockchainIDA.String())
-	chainARPCURI = httpToRPCURI(chainANodeURIs[0], blockchainIDA.String())
-	log.Info("Creating ethclient for blockchainA", "wsURI", chainAWSURI, "rpcURL, chainARPCURI")
-	chainARPCClient, err = ethclient.Dial(chainARPCURI)
-	Expect(err).Should(BeNil())
-
-	chainAIDInt, err = chainARPCClient.ChainID(context.Background())
-	Expect(err).Should(BeNil())
-
-	chainBWSURI := httpToWebsocketURI(chainBNodeURIs[0], blockchainIDB.String())
-	chainBRPCURI = httpToRPCURI(chainBNodeURIs[0], blockchainIDB.String())
-	log.Info("Creating ethclient for blockchainB", "wsURI", chainBWSURI)
-	chainBWSClient, err = ethclient.Dial(chainBWSURI)
-	Expect(err).Should(BeNil())
-	chainBRPCClient, err = ethclient.Dial(chainBRPCURI)
-	Expect(err).Should(BeNil())
-
-	chainBIDInt, err = chainBRPCClient.ChainID(context.Background())
-	Expect(err).Should(BeNil())
-	log.Info("Finished setting up e2e test subnet variables")
-
-	log.Info("Deploying Teleporter contract to subnets")
-	// Read in the Teleporter contract information
-	teleporterContractAddress = common.HexToAddress(readHexTextFile("./tests/UniversalTeleporterMessengerContractAddress.txt"))
-	teleporterDeployerAddress := common.HexToAddress(readHexTextFile("./tests/UniversalTeleporterDeployerAddress.txt"))
-	teleporterDeployerTransaction := readHexTextFile("./tests/UniversalTeleporterDeployerTransaction.txt")
-
-	nonceA, err := chainARPCClient.NonceAt(ctx, fundedAddress, nil)
-	Expect(err).Should(BeNil())
-
-	nonceB, err := chainBRPCClient.NonceAt(ctx, fundedAddress, nil)
-	Expect(err).Should(BeNil())
-
-	gasTipCapA, err := chainARPCClient.SuggestGasTipCap(context.Background())
-	Expect(err).Should(BeNil())
-	gasTipCapB, err := chainBRPCClient.SuggestGasTipCap(context.Background())
-	Expect(err).Should(BeNil())
+// Define the Relayer before and after suite functions.
+var _ = ginkgo.BeforeSuite(setupSuite)
 
-	baseFeeA, err := chainARPCClient.EstimateBaseFee(context.Background())
-	Expect(err).Should(BeNil())
-	gasFeeCapA := baseFeeA.Mul(baseFeeA, big.NewInt(relayerEvm.BaseFeeFactor))
-	gasFeeCapA.Add(gasFeeCapA, big.NewInt(relayerEvm.MaxPriorityFeePerGas))
+var _ = ginkgo.AfterSuite(teleporterTestUtils.TearDownNetwork)
 
-	baseFeeB, err := chainBRPCClient.EstimateBaseFee(context.Background())
+// Sets up the warp-enabled network and deploys the teleporter contract to each of the subnets
+func setupSuite() {
+	teleporterTestUtils.SetupNetwork(warpGenesisFile)
+	teleporterContractAddress := common.HexToAddress(testUtils.ReadHexTextFile("./tests/utils/UniversalTeleporterMessengerContractAddress.txt"))
```

Comment:
doesnt need to be in this PR, but thinking down the line how we can avoid manually updating these three files.